src/dataframe/dataframe.jl

```diff
@@ -1806,3 +1806,94 @@ function _replace_columns!(df::DataFrame, newdf::DataFrame)
     copy!(index(df).lookup, index(newdf).lookup)
     return df
 end
+
+expandgrid(; kwargs...) = isempty(kwargs) ? DataFrame() : expandgrid(kwargs...)
+
+expandgrid(pairs::Pair{<:AbstractString, <:Any}...) =
+    expandgrid((Symbol(k) => v for (k,v) in pairs)...)
+
+"""
+    expandgrid(pairs::Pair...)
+    expandgrid(kwargs...)
+
+Create a `DataFrame`` from all combinations of the passed arguments.
+
+It is allowed to pass  a list of `Pair`s as positional
+arguments, or a list of keyword arguments. Each `Pair`` is considered
+to represent a column name to column values to expand mapping.
+Column name must be a `Symbol` or string. All passed column names must be unique.
+
+Column value can be a vector which is consumed as is or an object of any other
+type (except `AbstractArray`). In the latter case the passed value is treated
+as having length one for expansion. As a particular rule values stored in a `Ref`
+or a `0`-dimensional `AbstractArray` are unwrapped and treated as having length one.
+
+`DataFrame` can store only columns that use 1-based indexing. Attempting
+to store a vector using non-standard indexing after `repeat` is called on it
+will raise an error.
```

Comment:
I don't understand this part. What is the reason why this should be mentioned in this docstring in particular?

Reply:
you could pass a vector that does not use `1`-based indexing to `expandgrid`. Then internally we call `repeat` on it. If the result of such `repeat` produces a non 1-based vector we will throw an error.

I could omit this comment as this error is caused by how `DataFrame` works and is not related to the `expandgrid` directly - if you think it is not needed to add this comment here.

Reply:
I imagine that in general array types for which `repeat` returns a non-1-based indexed object are also non-1-based themselves, right? If so, the error that we throw should be explicit enough for users. If we mention this here, we would have to mention it in many other docstrings.

Reply:
OK - removed.